rspirv/dr/constructs.rs

```diff
@@ -135,6 +132,70 @@ impl Module {
             .chain(&self.annotations)
             .chain(&self.types_global_values)
     }
+
+    /// Returns a mut iterator over all global instructions.
+    pub fn global_inst_iter_mut(&mut self) -> impl Iterator<Item = &mut Instruction> {
+        self.capabilities
+            .iter_mut()
+            .chain(&mut self.extensions)
+            .chain(&mut self.ext_inst_imports)
+            .chain(&mut self.memory_model)
+            .chain(&mut self.entry_points)
+            .chain(&mut self.execution_modes)
+            .chain(&mut self.debugs)
+            .chain(&mut self.annotations)
+            .chain(&mut self.types_global_values)
+    }
+
+    /// Returns a iterator over all instructions.
+    pub fn all_inst_iter(&self) -> impl Iterator<Item = &Instruction> {
+        self.capabilities
+            .iter()
+            .chain(&self.extensions)
+            .chain(&self.ext_inst_imports)
+            .chain(&self.memory_model)
+            .chain(&self.entry_points)
+            .chain(&self.execution_modes)
+            .chain(&self.debugs)
+            .chain(&self.annotations)
+            .chain(&self.types_global_values)
```

Comment:
```suggestion
        self.global_inst_iter()
```

Can't do this for `all_inst_iter_mut` though because `self` would be mutably borrowed for the duration of the result of `global_inst_iter_mut` and can't be mutably borrowed again for `self.functions.iter_mut()` :disappointed:

Reply:
I kept it duplicated because I think it's easier to keep them in sync that way - rather then having to have a special `mut` version.

Reply:
Makes sense; I was hoping you had an idea/solution for the mutability problem. Makes me wonder why borrowing `self` mutable multiple times for every argument to `chain()` is fine, while invoking a function (that should merely propagate the same "mutable lifetime scope") is not.

Reply:
It makes sense now: The compiler understands that distinct members of `self` are borrowed in the `.chain()` case. When the function is called however, `self` in its entirety is borrowed mutably.
Duplicating one of the `.chain()` invocations results in a compiler error: that field is already mutably borrowed.

I don't think there's an easy solution to this, let's leave the duplication in place.